cli/flox/src/commands/mod.rs

```diff
@@ -886,34 +888,114 @@ impl UninitializedEnvironment {
             },
         }
     }
-}
 
-impl Display for UninitializedEnvironment {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    /// If the environment is remote or managed, the name of the owner
+    pub fn owner(&self) -> Option<&EnvironmentOwner> {
         match self {
-            UninitializedEnvironment::DotFlox(DotFlox {
-                path,
-                pointer: EnvironmentPointer::Managed(managed_pointer),
-            }) => {
-                write!(
-                    f,
-                    "{}/{} at {}",
-                    managed_pointer.owner,
-                    managed_pointer.name,
-                    path.to_string_lossy(),
-                )
-            },
-            UninitializedEnvironment::DotFlox(DotFlox {
-                path,
-                pointer: EnvironmentPointer::Path(path_pointer),
-            }) => {
-                write!(f, "{} at {}", path_pointer.name, path.to_string_lossy())
-            },
-            UninitializedEnvironment::Remote(pointer) => {
-                write!(f, "{}/{} (remote)", pointer.owner, pointer.name,)
+            UninitializedEnvironment::DotFlox(DotFlox { pointer, .. }) => pointer.owner(),
+            UninitializedEnvironment::Remote(pointer) => Some(&pointer.owner),
+        }
+    }
+
+    /// The name of the environment
+    pub fn name(&self) -> &EnvironmentName {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { pointer, .. }) => pointer.name(),
+            UninitializedEnvironment::Remote(pointer) => &pointer.name,
+        }
+    }
+
+    /// Returns true if the environment is in the current directory
+    pub fn is_current_dir(&self) -> Result<bool> {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { path, .. }) => {
+                let current_dir = std::env::current_dir()?;
+                let is_current = current_dir.canonicalize()? == path.canonicalize()?;
+                Ok(is_current)
             },
+            UninitializedEnvironment::Remote(_) => Ok(false),
+        }
+    }
+
+    /// Returns the path to the environment if it isn't remote
+    #[allow(dead_code)]
+    pub fn path(&self) -> Option<&Path> {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { path, .. }) => Some(path),
+            UninitializedEnvironment::Remote(_) => None,
+        }
+    }
+
+    /// Returns true if the environment is a managed environment
+    pub fn is_managed(&self) -> bool {
+        match self {
+            UninitializedEnvironment::DotFlox(DotFlox { pointer, .. }) => pointer.owner().is_some(),
+            UninitializedEnvironment::Remote(_) => false,
         }
```

Comment:
```suggestion
        suggestion: use `matches!` or at least match on the type rather than a property of the variant.
        matches!(
            self,
            UninitializedEnvironment::DotFlox(DotFlox { pointer: EnvironmentPointer::Managed(_), .. }
        )
```

Reply:
Fixed